src/components/tile-manager/tile.ts

```diff
@@ -291,42 +292,57 @@ export default class IgcTileComponent extends EventEmitterMixin<
     });
   }
 
-  private handleDragStart(e: DragEvent) {
+  private async assignDragImage(e: DragEvent) {
     const rect = this.getBoundingClientRect();
     const offsetX = e.clientX - rect.left;
     const offsetY = e.clientY - rect.top;
+    const compStyles = getComputedStyle(this);
 
-    e.dataTransfer!.setDragImage(this._tileContent, offsetX, offsetY);
+    this._dragImage = this.cloneNode(true) as HTMLElement;
+    Object.assign(this._dragImage.style, {
+      width: compStyles.width,
+      height: compStyles.height,
+    });
+    this._dragImage.style.position = 'absolute';
+    this._dragImage.style.top = '-99999px';
+    this._dragImage.style.left = '-99999px';
+
+    document.body.append(this._dragImage);
+
+    e.dataTransfer!.setDragImage(this._dragImage, offsetX, offsetY);
     e.dataTransfer!.effectAllowed = 'move';
+  }
+
+  private async handleDragStart(e: DragEvent) {
+    await this.assignDragImage(e);
 
     this.emitEvent('tileDragStart', { detail: this });
-    this._isDragging = true;
+    this._dragGhost = this.ghostFactory();
```

Comment:
What's the idea behind this? We need an additional _ghost_ element for?

Reply:
The additional ghost is shown on the tiles' place, showing a visual clue of the current tile's position (or, in swap mode, the position where the drag started). I believe it came up as a design requirement and a solution to add some transition to the dragging operation.